### PR TITLE
Refactor some extensions-related code

### PR DESCRIPTION
**Pull request by britto**
Opportunistic refactoring from the work on #164. Concentrated config-reading code inside `get_all_modules`, optimized `cal_extensions` to do fewer iterations and removed a few unreachable/redundant code paths.

### CODE REVIEW

**Comment by whatyouhide** [comment]
@britto this can be rebased on the main branch :)

**Comment by britto** [comment]
Thank you! Done.

File: lib/protobuf/extension.ex [hunk 1]
```diff
@@ -114,18 +114,12 @@ defmodule Protobuf.Extension do
...
-      end
-    end)
-    |> Enum.map(fn mod ->
-      {mod, mod.__protobuf_info__(:extension_props)}
-    end)
-    |> Enum.reject(fn {_mod, props} -> is_nil(props) end)
-    |> Enum.each(fn {mod, props} ->
-      Enum.each(props.extensions, fn {_, ext} ->
+    for mod <- mods,
+        to_string(mod) =~ ~r/\.PbExtension$/,
```
Comment (whatyouhide) [inline lib/protobuf/extension.ex:118]: Two ideas, not blocking:

```suggestion
        mod |> Module.split() |> List.last() == "PbExtension"
```

or

```elixir
String.ends_with?(inspect(mod), ".PbExtension")
```

I think both solutions are a bit easier to read and more solid (that is, I am always always always scared of regexes) 😄

Comment (whatyouhide) [reply lib/protobuf/extension.ex:118]: I'll merge this PR, maybe we can do this in a future refactor :)

**Review by whatyouhide — APPROVED** [review]

**Pull request merged**